index.js

```diff
@@ -51,19 +52,26 @@ Auth.prototype.getAuthClient = function (callback) {
     var googleAuth = new GoogleAuth();
     var keyFile = config.keyFilename || config.keyFile;
 
-    if (config.credentials || keyFile && path.extname(keyFile) === '.json') {
-      var json = config.credentials;
+    if (config.credentials) {
+      googleAuth.fromJSON(config.credentials, addScope);
+    } else if (keyFile) {
+      keyFile = path.resolve(process.cwd(), keyFile);
 
-      if (!json) {
-        json = require(path.resolve(process.cwd(), keyFile));
-      }
+      fs.readFile(keyFile, function (err, contents) {
```

Comment:
Should this be `fs.readFile(keyFile, 'utf8', function (err, contents)`? Or does `JSON.parse` work on raw buffers?

https://nodejs.org/api/fs.html#fs_fs_readfile_file_options_callback